sys/userlib/src/critical_section.rs

```diff
@@ -0,0 +1,29 @@
+//! An implementation to support the `critical-section` crate within a Hubris
+//! user task.
+//!
+//! Hubris is careful never to introduce non-local or asynchronous control flow
+//! into a program, and doesn't have threads. This means that, within the
+//! context of a task, we don't actually need to generate any code to implement
+//! a critical section -- they happen naturally.
```

Comment:
_very_ tiny nit: in markdown, an em dash is `---`; i believe a `--` just renders as two en dashes.
```suggestion
//! a critical section --- they happen naturally.
```

Reply:
Fixed!